src/main.lua

```diff
@@ -418,8 +418,47 @@ local function addPrimaryNameRequestData(ioEvent, primaryNameResult)
 	addPrimaryNameCounts(ioEvent)
 end
 
+local function assertTableBytesLowerThan(tbl, max_bytes)
+	local seen = {}
+	local current_size = 0
+
+	local function sizeof(val)
+		if current_size > max_bytes then
+			error("Data size exceeds " .. max_bytes .. " bytes")
+		end
+
+		local t = type(val)
+		if t == "string" then
+			current_size = current_size + #val
+		elseif t == "number" or t == "boolean" then
+			current_size = current_size + 8 -- Approximate size for numbers/booleans
+		elseif t == "table" and not seen[val] then
```

Comment:
I'd be pretty surprised if the json deserializer is smart enough to save on table references in this way but nice thoroughness here.